### PR TITLE
NC | NSFS | IAM Service - Input Validation - `MaxItems`

### PR DESCRIPTION
**Pull request by shirady**
### Explain the changes
1. Parse the `max-items` to the number and test the new function.
2. Add JSDoc to the file `iam_utils`.
3. Uncomment a line that we accidentally comment in the test file `test_iam_utils.test.js`. 

### Issues: Fixed #xxx / Gap #xxx
1. I had a gap in PR #8172 in which I couldn't parse the value inside `max-items` from the request body (see [this comment](https://github.com/noobaa/noobaa-core/pull/8172#discussion_r1667615197)).
2. No pagination in `list-users` and `list_access_keys` implementations.

#### Unit Tests
Please run:
1. `npx jest test_iam_utils.test.js`

#### Manual Tests
1. Create the root user account with the CLI: `sudo node src/cmd/manage_nsfs account add --name shira-1001 --new_buckets_path /tmp/nsfs_root1 --access_key <access-key> --secret_key <secret-key> --uid <uid> --gid <gid>`
Note: before creating the account need to give permission to the `new_buckets_path`: `chmod 777 /tmp/nsfs_root1`.
3. Start the NSFS server with: `sudo node src/cmd/nsfs --debug 5 --https_port_iam 7005`
Note: before starting the server please add this line: `process.env.NOOBAA_LOG_LEVEL = 'nsfs';` in the endpoint.js (before the condition `if (process.env.NOOBAA_LOG_LEVEL) {`)
4. Create the alias for IAM service: `alias s3-nc-user-1-iam='AWS_ACCESS_KEY_ID=<access-key> AWS_SECRET_ACCESS_KEY=<secret-key> aws --no-verify-ssl --endpoint-url https://localhost:7005'`.
5. Use AWS CLI to create an IAM accounts: 
 user1: `s3-nc-user-1-iam iam create-user --user-name Bob` + `s3-nc-user-1-iam iam create-access-key --user-name Robert`
6. List access keys using `max-items` flag: `s3-nc-user-1-iam iam list-access-keys --max-items 1 --no-paginate --user-name Robert`
Check in the logs that it parsed as number: `core.endpoint.iam.ops.iam_list_access_keys:: IAM LIST ACCESS KEYS { marker: undefined, max_items: 1, username: ‘Robert’ }`
7.  List users using `max-items` flag: `s3-nc-user-1-iam iam list-users --max-items 7 --no-paginate`
Check in the logs that it parsed as number: `core.endpoint.iam.ops.iam_list_users:: IAM LIST USERS { marker: undefined, max_items: 7, iam_path_prefix: undefined }`
Note: to see the value of `--max-items` parsed I added the flag `--no-paginate`.

- [ ] Doc added/updated
- [X] Tests added

### CODE REVIEW

File: src/endpoint/iam/iam_utils.js [hunk 1]
```diff
@@ -82,6 +82,34 @@ function check_iam_path_was_set(iam_path) {
     return iam_path && iam_path !== IAM_DEFAULT_PATH;
 }
 
+/**
+ * parse_max_items converts the input to the needed type
+ * @param {any} input_max_items
+ */
+function parse_max_items(input_max_items) {
+    if (_.isUndefined(input_max_items)) return;
```
Comment (guymguym) [inline src/endpoint/iam/iam_utils.js:90]: no need for this function, just compare with x===undefined

File: src/endpoint/iam/iam_utils.js [hunk 2]
```diff
@@ -82,6 +82,34 @@ function check_iam_path_was_set(iam_path) {
...
+    const input_type = 'number';
+    const parameter_name = 'MaxItems';
+    if (typeof parameter_name !== input_type) {
```
Comment (guymguym) [inline src/endpoint/iam/iam_utils.js:93]: you meant to use typeof input not of the parameter name... but even so, i dont think this type check makes sense as the input will be of type string and will be parsed to integer, so just pass it to Number() and compare to NaN.

**Review by guymguym — CHANGES_REQUESTED** [review]

File: src/endpoint/iam/iam_utils.js (hunk 1 above)
Comment (shirady) [reply src/endpoint/iam/iam_utils.js:90]: Changed, thanks.

File: src/endpoint/iam/iam_utils.js (hunk 2 above)
Comment (shirady) [reply src/endpoint/iam/iam_utils.js:93]: Yes, I meant, changing, thanks.

File: src/endpoint/iam/iam_utils.js [hunk 3]
```diff
@@ -82,15 +82,47 @@ function check_iam_path_was_set(iam_path) {
...
 
+/**
+ * parse_max_items converts the input to the needed type
+ * assuming that we've got only sting type
+ * @param {any} input_max_items
+ */
+function parse_max_items(input_max_items) {
+    if (input_max_items === undefined) return;
+    const input_type = 'number';
+    const parameter_name = 'MaxItems';
```
Comment (guymguym) [inline src/endpoint/iam/iam_utils.js:93]: these consts seem to be unneeded

**Review by guymguym — APPROVED** [review]

File: src/endpoint/iam/iam_utils.js (hunk 3 above)
Comment (shirady) [reply src/endpoint/iam/iam_utils.js:93]: I'm using them in the error message (and prefer a template string).

**Pull request merged**